### ISSUE TITLE
Multiple warnings "W/art: Long monitor contention with owner pool-1-thread-1 .."

### ISSUE THREAD

**Issue by itaishalom** [couchbase/couchbase-lite-android#1634]
On 2.0.0-382
On automated tests:

```
03-01 18:57:55.441 6414-6452/com.example.itai.couchbasetest W/art: Long monitor contention with owner main (6414) at com.couchbase.lite.ResultSet com.couchbase.lite.ResultSet.refresh()(ResultSet.java:144) waiters=1 in com.couchbase.lite.Document com.couchbase.lite.Database.save(com.couchbase.lite.Document, boolean) for 323ms
03-01 18:57:55.442 6414-6414/com.example.itai.couchbasetest W/art: Long monitor contention with owner pool-1-thread-1 (6453) at void com.couchbase.lite.Database.postDatabaseChanged()(Database.java:1070) waiters=1 in com.couchbase.lite.ResultSet com.couchbase.lite.ResultSet.refresh() for 317ms
03-01 18:57:55.447 6414-6414/com.example.itai.couchbasetest I/LiteCore [SQL]: Created prerecorded query enum with 3 rows (120 bytes) in 4.217ms
```  
One update per second (On more then that it happens much faster) . Multiple warnings and no response from change() function.
 Emulator is: Samsung Galaxy S8+
 Running android 7.1.1 x86
 2GB Ram
 128 VM Heap

The setup:

One app with Content provider. On each update called, the app pulls from the db a document by id and save it with a new hash map (may be a big hash map with nested hash maps etc').
 There are three to four liveQuery alive, at least two of them are relevant to the updates.
 There are 5 updates each second from another app ( so another thread).
 From tests, each update takes at most 25 milliseconds (from contentprovider and after db.save(doc) )
 Hope it helps in the meantime

**Comment by itaishalom** [comment]
Note: If we run 5 updates per second for a while and these warnings comes out,
even if we slow down the updates for 1 per second, the system won't recover , there's something that is locked and no responses from the live query

**Comment by hideki** [comment]
Hi @itaishalom,
Could you please retest this with `2.0.0-401` too? 
http://mobile.maven.couchbase.com/feeds/dev//com/couchbase/lite/couchbase-lite-android/2.0.0-401

**Comment by itaishalom** [comment]
Hi, tested again and the warnings remains, when it happens no change() is called

**Comment by hideki** [comment]
@itaishalom,
I believe one of the database operations blocks `postDatabaseChanged()` method.
https://github.com/couchbase/couchbase-lite-android/blob/master/shared/src/main/java/com/couchbase/lite/AbstractDatabase.java#L993-L994
Currently, we can not reproduce the issue. We need your help to reproduce the issue.

**Comment by itaishalom** [comment]
I'll check how can I help you with that

**Comment by itaishalom** [comment]
Hi so, 
Server - https://github.com/itaishalom/CouchBase/tree/newTestUpdate

Client - https://github.com/itaishalom/couchbaseClientTest/tree/updateTest

Please test it on the emulator. I am currently OOO until Sunday morning (it's weekend now).
I couldn't manage to reproduce it with this code because I don't have emulator here but maybe you could. If not, please update me and we will see what can we do.

**Comment by hideki** [comment]
Hi @itaishalom,
Thank you for the sample application.
I can not reproduce the `W/art: Long monitor contention with owner...` warning with the sample.
Could you please provide additional information?
Thanks!

**Comment by hideki** [comment]
@itaishalom, It is obvious thread is blocked for 1sec or more. Without reproducing this issue. fixing is challenging. Please share the information as much as you can. Thanks!

**Comment by itaishalom** [comment]
I'll make a further investigation and contact you.

**Comment by hideki** [comment]
Hi @itaishalom 
This issue might be caused by same reason with issue #1693.
Could you please check if build 2.0.0-457 fix this issue too?
Thanks!
http://mobile.maven.couchbase.com/feeds/dev//com/couchbase/lite/couchbase-lite-android/2.0.0-457

**Comment by itaishalom** [comment]
I think it might be related to the dblistenertokens. Thanks

**Comment by hideki** [comment]
Hi @itaishalom,
Do you have a chance to retest this issue with using build 2.0.0-457?

**Comment by itaishalom** [comment]
Version 457 did wonderful. We can't rerun this exact same test but we ran others and the issue appeared to be resolved so I will close this ticket for now.
Thanks again.

**Comment by hideki** [comment]
Hi @itaishalom,
Thank you very much for your all contributions!

**Comment by hideki** [comment]
I will remove `known-issue` tag because this issue was addressed with the build 2.0.0-457.
